packages/admin-panel-server/src/viz-builder/dashboardVisualisation/DashboardVisualisationCombiner.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { DashboardItem, Report } from '../types';
+
+export class DashboardVisualisationCombiner {
```

Comment:
I've separated the logic that combines dashboardItem + report => visualisation to this separate class so that we don't have to cramp all the logic in the `DashboardVisualisationExtractor` class

Reply:
Nice and clean